### PR TITLE
Issue #DP-93: Update, Delete and Modify project

### PR DESCRIPTION
**Pull request by ahghatol**

### CODE REVIEW

File: src/app/client/src/app/modules/program/components/create-program/create-program.component.ts [hunk 1]
```diff
@@ -108,6 +108,7 @@ export class CreateProgramComponent implements OnInit, AfterViewInit {
   public programConfig: any;
   public disableCreateProgramBtn = false;
   public showLoader = true;
+  public issourcingOrgAdmin = false;
```
Comment (twsvaishali) [inline src/app/client/src/app/modules/program/components/create-program/create-program.component.ts:112]: In the constructor only.. we need to redirect the user to home if he is not sourcing admin

File: src/app/client/src/app/modules/program/components/create-program/create-program.component.ts [hunk 2]
```diff
@@ -746,6 +750,11 @@ export class CreateProgramComponent implements OnInit, AfterViewInit {
   }
 
   updateProgram($event: MouseEvent) {
+    if (!this.issourcingOrgAdmin) {
```
Comment (twsvaishali) [inline src/app/client/src/app/modules/program/components/create-program/create-program.component.ts:753]: @ahghatol These checks needs to be applied everywhere then. We will have some generic solution. This wont work.

File: src/app/client/src/app/modules/shared-feature/components/program-list/program-list.component.ts [hunk 3]
```diff
@@ -110,6 +113,11 @@ export class ProgramListComponent implements OnInit {
   }
 
   setDelete(program, index) {
+    if (!this.issourcingOrgAdmin) {
```
Comment (twsvaishali) [inline src/app/client/src/app/modules/shared-feature/components/program-list/program-list.component.ts:116]: Same   here as well @ahghatol

File: src/app/client/src/app/modules/shared-feature/components/program-list/program-list.component.ts [hunk 4]
```diff
@@ -154,6 +162,11 @@ export class ProgramListComponent implements OnInit {
   }
 
   deleteProject($event: MouseEvent){
+    if (!this.issourcingOrgAdmin) {
```
Comment (twsvaishali) [inline src/app/client/src/app/modules/shared-feature/components/program-list/program-list.component.ts:165]: here as well

File: src/app/client/src/app/modules/shared-feature/components/program-list/program-list.component.ts [hunk 5]
```diff
@@ -550,12 +567,10 @@ export class ProgramListComponent implements OnInit {
...
   editOnClick(program) {
-    if (this.iscontributeOrgAdmin) {
-      // if (this.activeMyProgramsMenu) {
-          return this.router.navigateByUrl('/sourcing/edit/' + program.program_id);
-      // }
+    if (this.issourcingOrgAdmin) {
+        return this.router.navigateByUrl('/sourcing/edit/' + program.program_id);
     } else {
-      return this.router.navigateByUrl('/sourcing/nominations/' + program.program_id);
+      return this.router.navigate(['home']);
```
Comment (twsvaishali) [inline src/app/client/src/app/modules/shared-feature/components/program-list/program-list.component.ts:573]: here as well

File: src/app/client/src/app/modules/program/components/create-program/create-program.component.ts (hunk 1 above)
Comment (ahghatol) [reply src/app/client/src/app/modules/program/components/create-program/create-program.component.ts:113]: This is not needed once we auth guard for this

File: src/app/client/src/app/modules/shared-feature/components/program-list/program-list.component.ts (hunk 3 above)
Comment (ahghatol) [reply src/app/client/src/app/modules/shared-feature/components/program-list/program-list.component.ts:116]: Here needed as we are showing program list to others users as well

File: src/app/client/src/app/modules/shared-feature/components/program-list/program-list.component.ts (hunk 4 above)
Comment (ahghatol) [reply src/app/client/src/app/modules/shared-feature/components/program-list/program-list.component.ts:165]: Needed here too

File: src/app/client/src/app/modules/program/components/create-program/create-program.component.ts (hunk 2 above)
Comment (ahghatol) [reply src/app/client/src/app/modules/program/components/create-program/create-program.component.ts]: Removed

**Pull request merged**